helix-term/src/commands.rs

```diff
@@ -1501,6 +1505,57 @@ fn switch_to_lowercase(cx: &mut Context) {
     });
 }
 
+pub fn scroll_page_and_cursor(cx: &mut Context, offset: usize, direction: Direction) {
+    let config = cx.editor.config();
+    let cursor_pos = cx.editor.cursor().0.unwrap();
+    let (view, doc) = current!(cx.editor);
+
+    let height = view.inner_height();
+    let scrolloff = config.scrolloff.min(height.saturating_sub(1) / 2);
+
+    let offset = match direction {
+        Direction::Forward => offset as isize,
+        Direction::Backward => -(offset as isize),
+    };
+
+    let doc_text = doc.text().slice(..);
+    let viewport = view.inner_area(doc);
+    let text_fmt = doc.text_format(viewport.width, None);
+    let mut annotations = view.text_annotations(doc, None);
+
+    (view.offset.anchor, view.offset.vertical_offset) = char_idx_at_visual_offset(
+        doc_text,
+        view.offset.anchor,
+        view.offset.vertical_offset as isize + offset,
+        0,
+        &text_fmt,
+        &annotations,
+    );
+
+    let movement = match cx.editor.mode {
+        Mode::Select => Movement::Extend,
+        _ => Movement::Move,
+    };
+
+    let mut offset = offset.abs() as usize;
```

Comment:
can we share the code between the normal scrolling and this scroll function to avoid code duplication. I have some further changes planned here and I don't want to do them twice

Reply:
[5a48de0](https://github.com/helix-editor/helix/pull/8015/commits/5a48de0800b4433cf1b8d9bb41bf74d1b5d3a11c)